models/classes/runner/QtiRunnerService.php

```diff
@@ -974,7 +974,8 @@ public function displayFeedbacks(RunnerServiceContext $context)
         /* @var TestSession $session */
         $session = $context->getTestSession();
 
-        return $session->getCurrentSubmissionMode() !== SubmissionMode::SIMULTANEOUS;
+        return $session->getCurrentSubmissionMode() !== SubmissionMode::SIMULTANEOUS
+            && $session->getCurrentAssessmentItemSession()->getItemSessionControl()->mustShowFeedback();
```

Comment:
```suggestion
return $this->isCurrentSubmissionModeSimultaneous($session)
            && $this->mustShowFeedback($session);
            
private function isCurrentSubmissionModeSimultaneous(TestSession $session): bool
    {
        return $session->getCurrentSubmissionMode() !== SubmissionMode::SIMULTANEOUS;
    }

    private function mustShowFeedback(TestSession $session): bool
    {
        return $session->getCurrentAssessmentItemSession()->getItemSessionControl()->mustShowFeedback();
    }
```

Reply:
I don't see much benefit in extracting it to a new method, to be honest. The second part for instance is one method call, creating another method to encapsulate this call seems to be overengineering. It's also important to note that this class is 2k lines long with 50+ methods.